zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/dmn/DecisionEvaluationEvaluteProcessor.java

```diff
@@ -54,6 +63,23 @@ public void processRecord(final TypedRecord<DecisionEvaluationRecord> command) {
     final DecisionEvaluationRecord record = command.getValue();
     final Either<Failure, PersistedDecision> decisionOrFailure = getDecision(record);
 
+    if (decisionOrFailure.isRight()) {
```

Comment:
💭 Just commenting here for clarity: It's more optimal to get the decision first, as it only performs one read/access to the (decision) state. If the decision exists, we can perform the authorization checks, which are more heavy, i.e. two accesses to the state (user and authorization).

Performing the authorization check first would be less performant, as the decision might not exist, so we would perform two state reads for nothing.